pkg/provider/provider.go

```diff
@@ -137,3 +161,44 @@ func convertToTypeSet(r *schema.Resource) *schema.Schema {
 		Elem: r,
 	}
 }
+
+// ValidateIAMVersion is a ValidateFunc for the "iam_version" field in the provider schema
+func ValidateIAMVersion(v interface{}, k string) ([]string, []error) {
+	// check that v is in iamVersionList
+	found := false
+	for _, version := range iamVersionList {
+		versn, ok := v.(string)
+		if !ok {
+			return []string{}, []error{fmt.Errorf("IAM version must be a string")}
+		}
+
+		if string(version) == versn {
+			found = true
+			break
+		}
+	}
+
+	// add error if not found
+	es := make([]error, 0)
+	if !found {
+		es = append(es, fmt.Errorf("IAM version must be one of %v", iamVersionList))
+	}
+
+	return []string{}, es
+}
+
+// ValidateServiceURL is a ValidateFunc for the "iam_service_url" field in the provider schema
+func ValidateServiceURL(v interface{}, k string) ([]string, []error) {
+	// check that v is a URL
+	_, ok := v.(string)
+	if !ok {
+		return []string{}, []error{fmt.Errorf("Service URL must be a string")}
+	}
+
+	_, err := url.ParseRequestURI(v.(string))
```

Comment:
Could use the result of the type assertion from line 193 here instead of casting again.

Reply:
Indeed we could.